components/selection/selection-header.js

```diff
@@ -0,0 +1,186 @@
+import '../overflow-group/overflow-group.js';
+import './selection-select-all.js';
+import './selection-select-all-pages.js';
+import './selection-summary.js';
+import { css, html, LitElement } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+import { SelectionObserverMixin } from './selection-observer-mixin.js';
+
+/**
+ * A header for selection components (e.g. list, table-wrapper) containing select-all, etc.
+ * @slot - Responsive container using `d2l-overflow-group` for `d2l-selection-action` elements
+ */
+export class SelectionHeader extends SelectionObserverMixin(RtlMixin(LocalizeCoreElement(LitElement))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether to render select-all and selection summary
+			 * @type {boolean}
+			 */
+			noSelection: { type: Boolean, attribute: 'no-selection' },
+			/**
+			 * Disables sticky positioning for the header
+			 * @type {boolean}
+			 */
+			noSticky: { type: Boolean, attribute: 'no-sticky' },
+			/**
+			 * Whether all pages can be selected
+			 * @type {boolean}
+			 */
+			selectAllPagesAllowed: { type: Boolean, attribute: 'select-all-pages-allowed' },
+			_hasActions: { state: true },
+			_scrolled: { type: Boolean, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				background-color: var(--d2l-selection-header-background-color, white);
+				display: block;
+				position: sticky;
+				top: 0;
+			}
+			:host([no-sticky]) {
+				background-color: transparent;
+				position: static;
+			}
+			.d2l-selection-header-shadow {
+				transition: box-shadow 200ms ease-out;
+			}
+			:host([_scrolled]) .d2l-selection-header-shadow {
+				background-color: var(--d2l-selection-header-background-color, white);
+				bottom: -4px;
+				box-shadow: 0 8px 12px -9px rgba(0, 0, 0, 0.3);
+				clip: rect(30px, auto, 200px, auto);
+				height: 40px;
+				position: absolute;
+				width: 100%;
+				z-index: -1;
+			}
```

Comment:
Note that I've kept the shadow styles the exact same as they were for `d2l-list-header`. I think these styles work fine as-is, but in the future we might want to tweak them a bit if any of these sizes/etc are specific to lists.

Reply:
Yeah these styles are very finicky. Jeff, Noah, and I spent a fair amount of time fiddling with them to get them working properly. I would hope we don't need different sets of shadow styles for different contexts.

Reply:
Cool. Yeah my concern is that our designs want a drop-shadow now for the sticky `<thead>`, and to accomplish that we'd need to apply the drop shadows to the thead or th's themselves, because it won't be valid to add extra divs in the table.